pkg/argocd/argocd.go

```diff
@@ -54,29 +55,38 @@ func AddClusterAddFlags(cmd *cobra.Command) (AddClusterCmd, error) {
 	return &addClusterImpl{root, args}, nil
 }
 
-func CheckAppSynced(ctx context.Context, f kube.Factory, ns, name string) (bool, error) {
-	rc, err := f.ToRESTConfig()
-	if err != nil {
-		return false, err
-	}
-
-	c, err := argocdcd.NewForConfig(rc)
-	if err != nil {
-		return false, err
-	}
+func GetAppSyncFn(revision string) kube.WaitFunc {
```

Comment:
I'd call it `WaitForAppSyncToRev`

Reply:
and you should probably add a comment for godoc

Reply:
It's a bit hacky, I admit. It's a function that accepts a version, and returns a WaitFunc. I just needed it so I have access to the revision in the WaitFunc, without adding this very specific parameter to the generic WaitFunc signature.

Reply:
So that is why the func is Get*Fn

Reply:
So `GetAppSyncWaitFunc`.

Reply:
And add a godoc comment on it.